### ISSUE TITLE
I do not understand how to build a project with support for websockets in the vs ?

### ISSUE THREAD

**Issue by heiheshang** [civetweb/civetweb#577]
I do not understand how to build a project with support for websockets in the vs ?

**Comment by bel2125** [comment]
vs = Visual Studio, right?

Did you try the existing Solution file: https://github.com/civetweb/civetweb/blob/master/VisualStudio/civetweb.sln ?

Did you set the USE_WEBSOCKET define (it's used often in the code, e.g. https://github.com/civetweb/civetweb/blob/master/src/civetweb.c#L2545)?

**Comment by heiheshang** [comment]
Yes,https://github.com/civetweb/civetweb/blob/master/VisualStudio/civetweb.sln I compiled the project.
I'm trying to determine 
#define USE_WEBSOCKET 1
getting an error
WEBSOCKET_OPCODE_TEXT undefined

**Comment by bel2125** [comment]
`WEBSOCKET_OPCODE_TEXT` is an old name and has been replaced by `MG_WEBSOCKET_OPCODE_TEXT`. It is still available when defining `MG_LEGACY_INTERFACE` - see https://github.com/civetweb/civetweb/blob/7a7d01ceab0a297520fc1efa26afa91fee3324e0/include/civetweb.h#L818

Where did you find the `WEBSOCKET_OPCODE_TEXT`? I thought it is replaced by `MG_WEBSOCKET_OPCODE_TEXT` everywhere in the CivetWeb repository.

**Comment by bel2125** [comment]
Duplicate: https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!topic/civetweb/ccRrsqbvWnU

**Comment by heiheshang** [comment]
all examples contain it

**Comment by heiheshang** [comment]
If you do not use_websocket then the examples are collected without any problems

**Comment by colmartofus** [comment]
I had the same problem recently. First a note on Visual Studio setup. Make sure you've defined `USE_WEBSOCKET`. In Visual Studio, this can be done in the `Project Properties/Configuration Properties/C C++/Preprocessor` then in the right pane use the `Preprocessor Definitions` field for `USE_SOCKET`.

If you're like me, you're using `embedded_cpp.cpp`. In this file, change both `WEBSOCKET_OPCODE_TEXT` usages to `MG_WEBSOCKET_OPCODE_TEXT`. The ones without the `MG_` shouldn't be there, I think. I believe the example file is just a little out of date. That's how I got my websockets to work. If that works, you could make a pull request to fix the examples.

Let me know if that worked.

**Comment by bel2125** [comment]
> all examples contain it

Thanks for sharing this information - I indeed forgot it for the C++ example.
I updated the C++ example now.

The examples in the "obsolete" folder are no longer actively maintained and may require a "MG_LEGACY_INTERFACE" define.

**Comment by bel2125** [comment]
Is there anything more we can do to help?
